app/src/main/java/com/amaze/filemanager/utils/OTGUtil.kt

```diff
@@ -146,20 +147,18 @@ object OTGUtil {
         context: Context,
         createRecursive: Boolean
     ): DocumentFile? {
-        val rootUriString = SingletonUsbOtg.getInstance().usbOtgRoot
-            ?: throw NullPointerException("USB OTG root not set!")
-
-        return getDocumentFile(path, rootUriString, context, OpenMode.OTG, createRecursive)
+        return getDocumentFile(path, context, OpenMode.OTG, createRecursive)
     }
 
     @JvmStatic
     fun getDocumentFile(
         path: String,
-        rootUri: Uri,
         context: Context,
         openMode: OpenMode,
         createRecursive: Boolean
     ): DocumentFile? {
+        val rootUri = SafRootHolder.uriRoot ?: return null
```

Comment:
This is wrong. Getting rootUri from Saf will render it useless to work with usbOtg. This is a general purpose method to make it work with origins of uri.